### PR TITLE
Fix for server_Address mismatch during initial conn & retry when CJWT is enabled

### PR DESCRIPTION
**Pull request by sadhyama**
When JWT is OFF -  parodus will connect with config server address during initial connection and retry cases.
When JWT is ON  -  parodus will connect with jwt_server_ip during initial connection and retry cases.
When JWT is ON & unable to get jwt_server_ip - parodus will connect with config server address.

### CODE REVIEW

File: src/connection.c [hunk 1]
```diff
@@ -168,9 +175,28 @@ int createNopollConnection(noPollCtx *ctx)
 				ParodusError("Error connecting to server\n");
 				ParodusError("RDK-10037 - WebPA Connection Lost\n");
 				// Copy the server address from config to avoid retrying to the same failing talaria redirected node
-				allow_insecure = parse_webpa_url (get_parodus_cfg()->webpa_url,
+				if (get_parodus_cfg()->acquire_jwt == 0)
```
Comment (bill1600) [inline src/connection.c]: This code seems to be repeated several times.
Wouldn't it be more readable if we made a function to encapsulate it ?

Comment (schmidtw) [reply src/connection.c]: It would/does.  I don't know that we should hold this change up for that vs. add it as an improvement.

Comment (bill1600) [reply src/connection.c]: OK, I'm fine with that.
I've got another change that needs to go in anyway.
But, can you please comment on my other comment, below.
If I've understood it right, doesn't this new code loop forever, trying to connect to the
jwt url, if there is one ?
What we want is for it to retry the jwt url N times, then revert to the default url.

**Comment by schmidtw** [comment]
Which lines are you concerned about retrying forever?

**Comment by schmidtw** [comment]
Shilpa helped me better understand.

Until the JWT expires, parodus should exclusively retry with that URL.  Once expired, the next reconnection will result in a restart of the entire process (including fetching the JWT if present).

**Pull request merged**